Adapt templates to work with new web app config

wtrocki opened the pull request.

wei-lee commented on openshift/datasync-showcase-amq.yml:
```diff
@@ -254,7 +220,7 @@ objects:
                   name: showcase-mqtt-credentials
                   key: MQTT_PASSWORD
-          image: ionic-showcase-server:latest
+          image: :latest
```
This needs to point to a valid image.

wei-lee commented on openshift/datasync-showcase-amq.yml:
```diff
@@ -263,10 +229,14 @@ objects:
+        - configMap:
+          name: webapp-config
     test: false
     triggers:
```
I think this section needs to be updated as well and remove the `imageChangeParams` part.

wtrocki commented:
Wrong update. Changing

wtrocki committed:
feat: Use configmap for the webapp

Motivation

Drop the builds of the showcase images.
Use configmap for dynamic configuration of the client side
Cleanup template from not needed elements

wtrocki committed:
fix: add image

wei-lee commented on openshift/datasync-showcase-amq.yml:
```diff
@@ -254,7 +218,7 @@ objects:
                   name: showcase-mqtt-credentials
                   key: MQTT_PASSWORD
-          image: ionic-showcase-server:latest
+          image: docker.io/aerogear/voyager-server-example-task:latest
```
can we use quay.io instead, if that's ok?

wei-lee reviewed (approved):
LGTM. One small suggestion is that to use quay.io instead of docker.io.

wtrocki committed:
fix: Replace to original image name

wtrocki committed:
fix: move to relative paths

darahayes commented on openshift/datasync-showcase-amq.yml:
```diff
@@ -263,21 +227,17 @@ objects:
           volumeMounts:
           - mountPath: /usr/src/app/files
             name: files-storage
+          - mountPath: /website/assets/config
```
the app isn't located at `/` inside the image. This path should be `/usr/src/app/website/assets/config`

wtrocki committed:
chore: update mount path

darahayes commented on openshift/datasync-showcase-amq.yml:
```diff
@@ -263,21 +227,17 @@ objects:
           persistentVolumeClaim:
             claimName: sync-files-storage
+        - configMap:
+          name: webapp-config
```
If you compare this with the docs: https://docs.openshift.com/container-platform/3.11/dev_guide/configmaps.html#configmaps-use-case-consuming-in-volumes

It's possible that whatever parses the templates is getting confused because you're missing a top level name variable. I would try changing it to this.

```
- name: webapp-config
  configMap:
    name: webapp-config
```

wtrocki committed:
chore: apply different format for config-maps

wtrocki committed:
feat: calculate websocket url
